### DISCUSSION TITLE
Feedback Request: Generate changelog based on Yaml files

### DISCUSSION THREAD

**Discussion by JDiPierro** [olivierlacan/keep-a-changelog#491] [Ideas]
Hey all,

I love the Keep-A-Changelog project and the format, but I hate dealing with merge conflicts due to the `Unreleased` section of the changelog. To solve this issue my team started keeping changelogs in Yaml files and then 'compiling' them into a Markdown file.

We achieve this with a tiny bit of code and a docker container in the build pipeline, check it out here: https://github.com/jdipierro/changelog-generator

I'd love feedback or pull requests with improvements :)

**Comment by aryelgois** [comment]
It is a good idea, but then there would be only one commit with the changes.. Being able to 'blame' the changelog and see who made something (and also having a summary of important commits) is a good feature

**Comment by jromero** [comment]
I really like the idea. Even with just two developers this conflict occurs so often. 

I'd vote to lose a blame on a changelog for not continuously having to update PRs.

**Comment by jromero** [comment]
In addition, it technically doesn't have to be yaml right? You can just piece out the changelog as one file per pr that gets parsed and merged (aka compiled). This way you can still keep the keep a changelog format.

Eg.

./changelog/pr-1.md

```
### Added
- Some feature
```

./changelog/pr-2.md

```
### Added
- Some other feature

### Fixed
- Some issue
```

Would result in:
```
### Added
- Some feature
- Some other feature

### Fixed
- Some issue

```

**Comment by lioman** [comment]
Gitlab.com uses a simmilar approch: https://about.gitlab.com/2018/07/03/solving-gitlabs-changelog-conflict-crisis/

**Comment by aryelgois** [comment]
@jromero said:

> I'd vote to lose a blame on a changelog for not continuously having to update PRs.

In a recent project, I actually went for that. My team would use the description area in GitLab to draft the changelog, and only during merge we would copy and paste into the Changelog file. That means we could not use the merge button from the web site, we had to merge locally. But hey, no conflicts in the file :D

This idea came after a lot of pain and conflict notification emails..

We also had a blockquote fixed at the bottom of the Unreleased section, that worked as some kind of marker for the diffing, reducing the conflicts even more. Something like:

```
> The changes above are not released yet.
```

This helped in old and long running branches that already had changes in the Changelog file, to avoid conflicts when merging from `main`
